lms/static/js/learner_dashboard/views/course_entitlement_view.js

```diff
@@ -0,0 +1,395 @@
+                     }.bind(this));
+                 },
+
+                 handleEnrollChange: function() {
+                     /*
+                     Handles enrolling in a course, unenrolling in a session and changing session.
+                     The new session id is stored as a data attribute on the option in the session-select element.
+                     */
+                     var isUnenrolling;
+
+                     // Grab the id for the desired session, an unenrollment event will return null
+                     this.currentSessionSelection = this.$('.session-select')
+                         .find('option:selected').data('session_id');
+                     isUnenrolling = !this.currentSessionSelection;
+
+                     // Do not allow for enrollment when button is disabled
+                     if (this.$('.enroll-btn-initial').hasClass('disabled')) return;
+
+                     // Display the indicator icon
+                     HtmlUtils.setHtml(this.$dateDisplayField,
+                         HtmlUtils.HTML('<span class="fa fa-spinner fa-spin" aria-hidden="true"></span>')
+                     );
+
+                     $.ajax({
+                         type: isUnenrolling ? 'DELETE' : 'POST',
+                         url: this.enrollUrl,
+                         contentType: 'application/json',
+                         dataType: 'json',
+                         data: JSON.stringify({
+                             course_run_id: this.currentSessionSelection
+                         }),
+                         statusCode: {
+                             201: _.bind(this.enrollSuccess, this),
+                             204: _.bind(this.unenrollSuccess, this)
+                         },
+                         error: _.bind(this.enrollError, this)
+                     });
+                 },
+
+                 enrollSuccess: function(data) {
+                     /*
+                     Update external elements on the course card to represent the now available course session.
+
+                     1) Show the change session toggle button.
+                     2) Add the new session's dates to the date field on the main course card.
+                     3) Hide the 'View Course' button to the course card.
+                     */
+                     var successIconEl = '<span class="fa fa-check" aria-hidden="true"></span>';
+
+                     // Update the model with the new session Id;
+                     this.entitlementModel.set({currentSessionId: this.currentSessionSelection});
+
+                     // Allow user to change session
+                     this.$triggerOpenBtn.removeClass('hidden');
+
+                     // Display a success indicator
+                     HtmlUtils.setHtml(this.$dateDisplayField,
+                         HtmlUtils.HTML(
+                             successIconEl + this.getAvailableSessionWithId(data.course_run_id).session_dates
```

Comment:
Why do the dates need to be marked as safe HTML?

Reply:
Could we just allow them to be escaped like everything else?

Reply:
Yeah, that works. I'll change it.